### PR TITLE
Add fishing spot display configuration

### LINKED ISSUE #781: An additional option for Fishing displays: Hide the fish names
- An additional option for Fishing displays: Hide the fish names, just show the active fishing spots. I am aware they show without text on the minimap, but would like it to be on the game screen aswell.

### PR DESCRIPTION
**Pull request by deathbeam**
- Add separate config option to display fishing spot tiles
- Add separate config option to display fishing spot names
- Make config option to display fishing spot icons be independant of
fishing spot names

Closes #781

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingConfig.java [hunk 1]
```diff
@@ -24,22 +24,39 @@
  */
 package net.runelite.client.plugins.fishing;
 
+import lombok.RequiredArgsConstructor;
 import net.runelite.client.config.Config;
 import net.runelite.client.config.ConfigGroup;
 import net.runelite.client.config.ConfigItem;
 
 @ConfigGroup("fishing")
-public interface FishingConfig extends Config
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingConfig.java]: This has to be public

**Comment by Nightfirecat** [comment]
Wiki should be updated if this is merged

**Comment by deathbeam** [comment]
Updated to do what I mentioned in previous comment

File: runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingSpotOverlay.java [hunk 2]
```diff
@@ -121,27 +122,39 @@ public Dimension render(Graphics2D graphics)
...
+					}
 				}
 			}
-			else
+
+			if (config.showSpotNames())
 			{
 				String text = spot.getName();
-				OverlayUtil.renderActorOverlay(graphics, npc, text, color.darker());
+				Point textLocation = npc.getCanvasTextLocation(graphics, text, npc.getLogicalHeight() + 40);
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingSpotOverlay.java]: Should the height always be incremented by 40? Is this to make space for the icon or for some other reason? (if so, we could check for that and not add the space if the icon is hidden)

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingSpotOverlay.java]: It is literally copied from the OverlayUtil, that is just where we display text for every actor text overlay

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingSpotOverlay.java]: Ok :man_shrugging:

**Pull request merged**